drivers/at86rf2xx/at86rf2xx.c

```diff
@@ -56,6 +56,9 @@ void at86rf2xx_setup(at86rf2xx_t *dev, const at86rf2xx_params_t *params)
     /* initialize device descriptor */
     dev->params = *params;
 #endif
+    /* generate EUI-64 and short address */
+    luid_get_eui64(&dev->netdev.long_addr);
+    luid_get_short(&dev->netdev.short_addr);
```

Comment:
```suggestion
    luid_get_eui64((eui64_t *) &dev->netdev.long_addr);
    luid_get_short((network_uint16_t *)&dev->netdev.short_addr);
```

Reply:
This won't work for now, as `eui64_t` has an alignment requirement of 64 bit, but `dev->netdev.long_addr` only has an alignment requirement of 8 bit. This cast would result in undefined behavior. E.g. on some ARM chips an unaligned access will result in hard faults..

The only options we have here are:

1. Use a temporary `eui64_t` and `memcpy()`, adding one unnecessary copy operation.
2. Change the API of `luid_get_eui64()` to accept a `void` Pointer as argument
3. Change the type of the member in `netdev_ieee802154_t::long_addr`.

(And the same for the short address.)

I don't think that we should go for 1. here. We have full control over our APIs and data types, so we should just make sure those are compatible with each other.

Reply:
I'm sorry I misread what you had explained before, and though that your latest push was removing the parts of your PR that depended on additional changes.

Reply:
I did indeed remove some parts: In the meantime, `luid_get()` was replaced with `luid_get_eui64()`, fixing one bug this PR originally targeted. Now, only the reset issue (new address after each reset) is left.

Reply:
Ok, I was naively suggesting the cast just because I had compiler errors.

Reply:
Let try to reach a consensus in #12713 then

Reply:
Now that I'm aware that those crazy byteorder types use `__attribute__((packed))`, I'm fine with the cast. (There is not alignment mismatch due to this.)